stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketUtil.java

```diff
@@ -256,6 +290,428 @@ public static ErrorValue createWebSocketError(WebSocketConstants.ErrorCode code,
+        }
+    }
+
+    /**
+     * Get the webSocket service.
+     * @param clientEndpointConfig a client endpoint config
+     * @param strand a strand
+     * @return webSocketService
+     */
+    public static WebSocketService getWebSocketService(MapValue<String, Object> clientEndpointConfig, Strand strand) {
+        Object clientService = clientEndpointConfig.get(WebSocketConstants.CLIENT_SERVICE_CONFIG);
+        if (clientService != null) {
+            BType param = ((ObjectValue) clientService).getType().getAttachedFunctions()[0].getParameterType()[0];
+            if (param == null || !((WebSocketConstants.WEBSOCKET_CLIENT_NAME.equals(
+                    param.toString())) || (WebSocketConstants.WEBSOCKET_FAILOVER_CLIENT_NAME.equals(
+                    param.toString())))) {
+                throw new WebSocketException("The callback service should be a WebSocket Client Service");
+            }
+            return new WebSocketService((ObjectValue) clientService, strand.scheduler);
+        } else {
+            return new WebSocketService(strand.scheduler);
+        }
+    }
+
+    /**
+     * Do the doReconnect when webSocket connection will be lost.
+     *
+     * @param connectionInfo a connection info
+     * @return if do the reconnection, return true
+     */
+    static boolean doReconnect(WebSocketOpenConnectionInfo connectionInfo) {
+        ObjectValue webSocketClient = connectionInfo.getWebSocketEndpoint();
+        RetryContext retryConnectorConfig = (RetryContext) webSocketClient.getNativeData(RETRY_CONFIG);
+        int interval = retryConnectorConfig.getInterval();
+        int maxInterval = retryConnectorConfig.getMaxInterval();
+        int maxAttempts = retryConnectorConfig.getMaxAttempts();
+        int noOfReconnectAttempts = retryConnectorConfig.getReconnectAttempts();
+        float backOfFactor = retryConnectorConfig.getBackOfFactor();
+        WebSocketService wsService = connectionInfo.getService();
+        WebSocketClientConnector clientConnector = (WebSocketClientConnector) webSocketClient.
+                getNativeData(CLIENT_CONNECTOR);
+
+        Date date = new Date();
+        SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
+        // Check the no of reconnection attempt equals with maximum reconnection attempt or not.
+        // If it isn't equal, call the initialiseWebSocketConnection()
+        // if it equals, return false
+        if (((noOfReconnectAttempts < maxAttempts) && maxAttempts > 0) || maxAttempts  == 0) {
+            retryConnectorConfig.setReconnectAttempts(noOfReconnectAttempts + 1);
+            logger.info(formatter.format(date.getTime()) + " " + RECONNECTING);
+            setCountDownLatch(getWaitTime(interval, maxInterval, backOfFactor, noOfReconnectAttempts));
+            establishWebSocketConnection(clientConnector, webSocketClient, wsService);
+            return true;
+        }
+        return false;
```

Comment:
Without returning a boolean whether it succeeded or not, let's check that here and print the info log which has been duplicated in many places?